### DISCUSSION TITLE
Skip autogenerate for an engine in multiple database

### DISCUSSION THREAD

**Discussion by 9bany** [sqlalchemy/alembic#1252] [Usage Questions] [answered]
**Describe the use case**

I set up Alembic with multiple databases, including PostgreSQL and ClickHouse. I want Alembic to autogenerate with the PostgreSQL database, but skip autogeneration for the ClickHouse.

**Databases / Backends / Drivers targeted**

**Example Use**

With ClickHouse: I want to use a raw query in database migration because my database is very difficult.

**Additional context**

- I used [clickhouse-sqlalchemy](https://github.com/xzkostyan/clickhouse-sqlalchemy) plugin but it is not compatible with Alembic and does not strongly support ClickHouse.
- [clickhouse-sqlalchemy](https://github.com/xzkostyan/clickhouse-sqlalchemy) raised a bug when I tried to use it with alembic. link to ->  https://github.com/xzkostyan/clickhouse-sqlalchemy/issues/250
**Have a nice day!**

**Comment by CaselIT** [comment]
Hi,

Can't you just configure alembic to only use the postgresql engine?

**Comment by 9bany** [comment]
I can't @CaselIT

**Reply by CaselIT** [reply]
I'm not sure what's the issue in doing this is. If you don't plan to use alembic for a db I can't see the point of making alembic run on that db.

In any case you may be able to filter what autogenerate looks as by using the options documented here https://alembic.sqlalchemy.org/en/latest/autogenerate.html#controlling-what-to-be-autogenerated

**Reply by 9bany** [reply, marked as answer]
I applied to create an alembic.ini file for each database engine that can work, but I hope my case can work with multiple database templates in Alembic, and I can configure it correctly with those.
Thanks for your support, @CaselIT
